x-pack/plugins/security_solution/scripts/run_cypress/parallel_serverless.ts

```diff
@@ -0,0 +1,466 @@
+      headers: {
+        Authorization: `ApiKey ${apiKey}`,
+      },
+    });
+    log.info(`Environment ${projectName} was successfully deleted!`);
+  } catch (error) {
+    log.error(`${error}`);
+  }
+}
+
+// Method to reset the credentials for the created environment.
+async function resetCredentials(
+  environmentId: string,
+  runnerId: string,
+  apiKey: string
+): Promise<Credentials | undefined> {
+  log.info(`${runnerId} : Reseting credentials`);
+  try {
+    const response = await axios.post(
+      `${BASE_ENV_URL}/api/v1/serverless/projects/security/${environmentId}/_reset-credentials`,
+      {},
+      {
+        headers: {
+          Authorization: `ApiKey ${apiKey}`,
+        },
+      }
+    );
+    return {
+      password: response.data.password,
+      username: response.data.username,
+    };
+  } catch (error) {
+    throw new Error(`${error}`);
+  }
+}
+
+// Wait until elasticsearch status goes green
+function waitForEsStatusGreen(esUrl: string, auth: string, runnerId: string): Promise<void> {
+  const fetchHealthStatusAttempt = async (attemptNum: number) => {
+    log.info(`Retry number ${attemptNum} to check if es is green.`);
+
+    const response = await axios.get(`${esUrl}/_cluster/health?wait_for_status=green&timeout=50s`, {
+      headers: {
+        Authorization: `Basic ${auth}`,
+      },
+    });
+
+    log.info(`${runnerId}: Elasticsearch is ready with status ${response.data.status}.`);
+  };
+  const retryOptions = {
+    onFailedAttempt: (error: Error | AxiosError) => {
+      if (error instanceof AxiosError && error.code === 'ENOTFOUND') {
+        log.info(
+          `${runnerId}: The elasticsearch url is not yet reachable. A retry will be triggered soon...`
+        );
+      }
+    },
+    retries: 50,
+    factor: 2,
+    maxTimeout: 20000,
+  };
+
+  return pRetry(fetchHealthStatusAttempt, retryOptions);
+}
+
+// Wait until Kibana is available
+function waitForKibanaAvailable(kbUrl: string, auth: string, runnerId: string): Promise<void> {
+  const fetchKibanaStatusAttempt = async (attemptNum: number) => {
+    log.info(`Retry number ${attemptNum} to check if kibana is available.`);
+    const response = await axios.get(`${kbUrl}/api/status`, {
+      headers: {
+        Authorization: `Basic ${auth}`,
+      },
+    });
+    if (response.data.status.overall.level !== 'available') {
+      throw new Error(`${runnerId}: Kibana is not available. Retrying in 20s...`);
+    } else {
+      log.info(`${runnerId}: Kibana status overall is ${response.data.status.overall.level}.`);
+    }
+  };
+  const retryOptions = {
+    onFailedAttempt: (error: Error | AxiosError) => {
+      if (error instanceof AxiosError && error.code === 'ENOTFOUND') {
+        log.info(`${runnerId}: The kibana url is not yet reachable. Retrying in 20s...`);
+      } else {
+        log.info(`${runnerId}: ${error}`);
+      }
+      throw new Error(`${runnerId} - ${error}`);
+    },
+    retries: 50,
+    factor: 2,
```

Comment:
It looks like we can have a delay before attempting to fetch the status. As far as I know an MKI env takes around 6 minutes to be up and running. A delay can be a minute or two or the other amount of time we certain about.

Additionally we can "play" with params like `factor`, `minTimeout` and `maxTimeout` to find an optimal approach. Most probably a linear or fixed attempt interval with a delay before will work better.

It doesn't have to be addresses in this PR, just a general though.